src/modules/job-info/guest_watch.c

```diff
@@ -607,32 +610,25 @@ static void guest_namespace_watch_continuation (flux_future_t *f, void *arg)
 {
     struct guest_watch_ctx *gw = arg;
     struct info_ctx *ctx = gw->ctx;
-    const char *s;
+    const char *event;
 
-    if (flux_rpc_get (f, &s) < 0) {
+    if (flux_job_event_watch_get (f, &event) < 0) {
         if (errno == ENOTSUP) {
-            /* Guest namespace has been removed.  If we have read
-             * events in the guest eventlog, we can assume the guest
-             * eventlog is done, we'll return ENODATA to the caller.
-             *
-             * If we have read no events, one of two possibilities exist:
-             * - the job was moved into the main namespace before we
-             *   began watching the guest namespace.
-             * - the guest eventlog is empty
-             *
-             * There's no way to be for certain, so we fallthrough to
-             * the main KVS namespace in both circumstances.
+            /* Guest namespace has been removed and eventlog has been
+             * moved to primary KVS namespace.  Fallthrough to primary
+             * KVS namespace to ensure that we have output all data to
+             * the user.  This is very small race possibility where
```

Comment:
Is the "very small race possibility" documenting a known race or describing code that fixes the race?   It's right on top of another comment block describing what looks like a fix for a different race.

Reply:
let me see how fixing up those comments.  There are really two races, and both are fixed in the same place.